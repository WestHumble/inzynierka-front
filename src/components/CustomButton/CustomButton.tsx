import { View, Text, StyleSheet, Pressable } from "react-native";
import React from "react";

const CustomButton = ({
  onPress,
  text,
  type = "PRIMARY",
  bgColor,
  fgColor,
  additionalStyles,
}) => {
  return (
    <Pressable
      onPress={onPress}
      style={[
        styles.container,
        styles[`container_${type}`],
        bgColor ? { backgroundColor: bgColor } : {},
        additionalStyles,
      ]}
    >
      <Text
        style={[
          styles.text,
          styles[`text_${type}`],
          fgColor ? { color: fgColor } : {},
        ]}
      >
        {text}
      </Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    width: "100%",
    padding: 15,
    marginVertical: 5,
    alignItems: "center",
    borderRadius: 5,
  },
  container_PRIMARY: {
    backgroundColor: "#F2B138",
  },
  container_SECONDARY: {
    backgroundColor: "#003f63",
  },
  container_TERTIARY: {
    backgroundColor: "transparent",
  },
  text: {
    fontWeight: "bold",
  },
  text_PRIMARY: {
    color: "white",
  },
  text_SECONDARY: {
    color: "black",
  },
  text_TERTIARY: {
    color: "#999999",
  },
});

export default CustomButton;
