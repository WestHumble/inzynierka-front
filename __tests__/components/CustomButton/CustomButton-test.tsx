import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from "../../../src/components/CustomButton";
import { expect, test } from '@jest/globals';

test('renders correctly', () => {
  const tree = renderer.create(<CustomButton
      text="Test"
      onPress={undefined}
      type="PRIMARY"
      bgColor={undefined}
      fgColor={undefined}
      additionalStyles={undefined}
      additionalStylesText={undefined}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});


test('renders with function', () => {
  const tree = renderer.create(<CustomButton
      text="Test"
      onPress={()=>{}}
      type="PRIMARY"
      bgColor={undefined}
      fgColor={undefined}
      additionalStyles={undefined}
      additionalStylesText={undefined}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});