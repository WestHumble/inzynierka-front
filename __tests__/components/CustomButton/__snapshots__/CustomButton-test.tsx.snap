// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 20,
        "height": 60,
        "justifyContent": "center",
        "marginVertical": 5,
        "padding": 0,
        "width": "100%",
      },
      {
        "backgroundColor": "#F2B138",
      },
      {},
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "fontWeight": "bold",
        },
        {
          "color": "white",
        },
        {},
        undefined,
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`renders with function 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 20,
        "height": 60,
        "justifyContent": "center",
        "marginVertical": 5,
        "padding": 0,
        "width": "100%",
      },
      {
        "backgroundColor": "#F2B138",
      },
      {},
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "fontWeight": "bold",
        },
        {
          "color": "white",
        },
        {},
        undefined,
      ]
    }
  >
    Test
  </Text>
</View>
`;
